app/controllers/pieces_controller.rb

```diff
@@ -30,15 +33,16 @@ def board_display_piece_query(row, column)
   def show_piece_td(row, column)
     @current_game = current_game
     find_piece = board_display_piece_query(row, column)
-    board_square = "<td class='x-position-'#{column}' "
-    board_square += "piece-id-data='#{piece_id(find_piece)}' "
-    board_square += "piece-type-data='#{piece_type(find_piece)}''>"
+    board_square = "<td data-x-position='#{column}'"
```

Comment:
more fixing of custom HTML5 data attributes (good reading if you don't know about these)